bigbluebutton-html5/imports/api/group-chat-msg/server/modifiers/clearGroupChatMsg.js

```diff
@@ -1,13 +1,32 @@
 import GroupChatMsg from '/imports/api/group-chat-msg';
 import Logger from '/imports/startup/server/logger';
+import addGroupChatMsg from '/imports/api/group-chat-msg/server/modifiers/addGroupChatMsg';
 
 export default function clearGroupChatMsg(meetingId, chatId) {
-  if (meetingId) {
-    return GroupChatMsg.remove({ meetingId }, Logger.info(`Cleared GroupChat (${meetingId})`));
-  }
+  const CHAT_CONFIG = Meteor.settings.public.chat;
+  const PUBLIC_CHAT_SYSTEM_ID = CHAT_CONFIG.system_userid;
+  const PUBLIC_GROUP_CHAT_ID = CHAT_CONFIG.public_group_id;
+  const CHAT_CLEAR_MESSAGE = CHAT_CONFIG.system_messages_keys.chat_clear;
 
   if (chatId) {
-    return GroupChatMsg.remove({ meetingId, chatId }, Logger.info(`Cleared GroupChat (${meetingId}, ${chatId})`));
+    GroupChatMsg.remove({ meetingId, chatId }, Logger.info(`Cleared GroupChat (${meetingId}, ${chatId})`));
+
+    const clearMsg = {
+      color: '0',
+      timestamp: Date.now(),
+      correlationId: `${PUBLIC_CHAT_SYSTEM_ID}-${Date.now()}`,
+      sender: {
+        id: PUBLIC_CHAT_SYSTEM_ID,
+        name: '',
+      },
+      message: CHAT_CLEAR_MESSAGE,
+    };
+
+    return addGroupChatMsg(meetingId, PUBLIC_GROUP_CHAT_ID, clearMsg);
+  }
+
+  if (meetingId) {
```

Comment:
Looks like you remove some messages on line 12 and then a lot more messages on line 29. Why is that? Aren't you removing the message you just inserted - `CHAT_CLEAR_MESSAGE`?

Reply:
No, because the `remove()` in line 29 will only be executed if the `chatId` parameter is missing, it's implemented this way because in https://github.com/bigbluebutton/bigbluebutton/blob/v2.2-dev/bigbluebutton-html5/imports/api/chat/server/modifiers/clearChats.js we had this function to clean all chats from a meeting that is called in https://github.com/bigbluebutton/bigbluebutton/blob/v2.2-dev/bigbluebutton-html5/imports/api/meetings/server/modifiers/removeMeeting.js#L17